node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -16,32 +16,51 @@ import org.bitcoins.core.p2p.TransactionMessage
 import org.bitcoins.core.p2p.MerkleBlockMessage
 import org.bitcoins.node.SpvNodeCallbacks
 import org.bitcoins.core.p2p.GetDataMessage
+import org.bitcoins.node.models.BroadcastAbleTransactionDAO
+import slick.jdbc.SQLiteProfile
+import org.bitcoins.node.config.NodeAppConfig
 
 /** This actor is meant to handle a [[org.bitcoins.node.messages.DataPayload]]
   * that a peer to sent to us on the p2p network, for instance, if we a receive a
   * [[HeadersMessage]] we should store those headers in our database
   */
 class DataMessageHandler(callbacks: SpvNodeCallbacks)(
     implicit ec: ExecutionContext,
-    appConfig: ChainAppConfig)
+    chainConf: ChainAppConfig,
+    nodeConf: NodeAppConfig)
     extends BitcoinSLogger {
 
-  val callbackNum = callbacks.onBlockReceived.length + callbacks.onMerkleBlockReceived.length + callbacks.onTxReceived.length
+  private val callbackNum = callbacks.onBlockReceived.length + callbacks.onMerkleBlockReceived.length + callbacks.onTxReceived.length
   logger.debug(s"Given $callbackNum of callback(s)")
 
   private val blockHeaderDAO: BlockHeaderDAO = BlockHeaderDAO()
+  private val txDAO = BroadcastAbleTransactionDAO(SQLiteProfile)
 
   def handleDataPayload(
       payload: DataPayload,
       peerMsgSender: PeerMessageSender): Future[Unit] = {
 
     payload match {
+      case getData: GetDataMessage =>
+        logger.debug(
+          s"Received a getdata message for inventories=${getData.inventories}")
+        getData.inventories.foreach { inv =>
+          logger.debug(s"Looking for inv=$inv")
+          txDAO.findByHash(inv.hash).map {
```

Comment:
I think we want to avoid database accesses for every single `GetDataMessage` we receive. I think something like this is superior 

```scala
        getData.inventories.foreach { inv =>
          logger.debug(s"Looking for inv=$inv")
          inv.typeIdentifier match {
            case TypeIdentifier.MsgTx =>
              txDAO.findByHash(inv.hash).map {
                case Some(tx) =>
                  peerMsgSender.sendTransactionMessage(tx.transaction)
                case None =>
                  logger.warn(
                    s"Got request to send data with hash=${inv.hash}, but found nothing")
              }
            case TypeIdentifier.MsgBlock | TypeIdentifier.MsgFilteredBlock |
                TypeIdentifier.MsgCompactBlock |
                TypeIdentifier.MsgFilteredWitnessBlock |
                TypeIdentifier.MsgWitnessBlock | TypeIdentifier.MsgWitnessTx =>
            //not sure what we want to do here on this PR...

            case unassigned: MsgUnassigned =>
              logger.warn(
                s"Received unassigned message we do not understand, msg=${unassigned}")
          }

        }
```